server/controllers/report.js

```diff
@@ -71,10 +105,38 @@ function list(req, res, next) {
  * GET /reports/archive/:uuid
  */
 function sendArchived(req, res, next) {
-  // @TODO This will have to factor in the type of report - report uuid can be looked up in `saved_report` table
-  let reportPath = path.resolve(path.join(__dirname, '../reports/'));
-  let extension = '.pdf';
-  let reportUuid = req.params.uuid;
+  lookupArchivedReport(req.params.uuid)
+    .then(report => {
+      res.sendFile(report.link);
+    })
+    .catch(next)
+    .done();
+}
 
-  res.sendFile(reportPath.concat('/', reportUuid, extension));
+
+/**
+ * @function deleteArchived
+ *
+ * @description
+ * Deletes a report from the server.  This cleans up both the record of the
+ * report stored in saved_report and the file stored on the disk.
+ *
+ * DELETE /reports/archive/:uuid
+ */
+function deleteArchived(req, res, next) {
+  let filePath;
+
+  lookupArchivedReport(req.params.uuid)
+    .then(report => {
+      filePath = report.link;
+      return db.exec('DELETE FROM saved_report WHERE uuid = ?;', [ db.bid(req.params.uuid) ]);
+    })
+    .then(() => {
+      fs.unlink(filePath, err => {
+        if (err) { return next(err); }
```

Comment:
I agree with your logic here. A cleanup cron job seems much safer and easier to automate than actively deleting corrupt references from the database. Storage is relatively cheap and it would take a great deal of persistent errors to have a significant impact. 

Thanks for the analysis, sounds good to me!